src/client/customers.rs

```diff
@@ -432,6 +437,203 @@ pub struct VoidInitiatedLedgerEntry {
     pub void_amount: serde_json::Number,
 }
 
+/// The view mode for a cost breakdown.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Deserialize_enum_str, Serialize_enum_str)]
+pub enum CostViewMode {
+    /// Provide results as an incremental day-by-day view.
+    #[serde(rename = "periodic")]
+    Periodic,
+    /// Provide results as cumulative totals since the start of the billing period.
+    #[serde(rename = "cumulative")]
+    Cumulative,
+}
+
+#[derive(Debug, Default, Clone)]
+struct CustomerCostParamsFilter<'a> {
+    timeframe_start: Option<OffsetDateTime>,
```

Comment:
do these need `#[serde(with = "time::serde::rfc3339")]` like other places in this file?

Reply:
They don't since Serde is not handling them. However, it did highlight that I needed to apply the formatter at the place these are converted to params.